### DISCUSSION TITLE
The 180s timeout for PHPAGI scripts

### DISCUSSION THREAD

**Discussion by jorikfon** [mikopbx/Core#44] [Q&A] [answered]
You know that we are big fans of PHPAGI.
For this year we prepared a lot of script and server templates to bring MIKO to a lot of customers...
Unfortunately, we have still or again the timer problem. I can do what ever to signal call processing - but - after 180s the script will be started again or next script will be called 
There must be found a way - to definitely terminate the call processing - if PHPAGI signals that.
After call is connected and established between caller and extension - we tell the script "hangup" ...  this is seems to be not recognized by your calling script - because after the timeout of 180s - the processing of the call continuous ...

The problem is here: (automatic generated extension.conf by MIKO)

 ```
 same => n,Set(M_TIMEOUT=180) 
  same => n,ExecIf($["${M_DIALSTATUS}" != "ANSWER"]?Dial(Local/000020201@internal-incoming/n,180,cTKg));
  same => n,Set(M_TIMEOUT=0)
  same => n,ExecIf($["${M_DIALSTATUS}" != "ANSWER"]?Dial(Local/201@internal/n,,cTKg)); default action
```

"g" - tells to go on with processing - if there is a HANGUP

So - solution for now seems to be -

*  changing extension.conf by hand
* Work-A-Round: use the script as "failover" - if no other extension is found - delegate to this script....

Maybe it would be good to make a simple little checkbox ( 😉))))))) to tell that this "rule" will end processing the call.

**Comment by boffart** [comment, marked as answer]
`$conf .= 'exten => _.!,1,ExecIf($["${MASTER_CHANNEL(M_TIMEOUT)}x" != "x"]?Set(TIMEOUT(absolute)=${MASTER_CHANNEL(M_TIMEOUT)}))' . " \n\t";`

Timeout sets to the channel generated by the command DIal.
The channel name is stored in a variable **MASTER_CHANNEL(M_TIMEOUT_CHANNEL)**

The timeout can be controlled, if in AGI invoke **answer**()

**Comment by boffart** [comment]
https://github.com/mikopbx/Core/blob/18615a3f4e9a393c63d178cde0f691eecb0dabe0/src/Core/Asterisk/agi-bin/clean_timeout.php#L26-L26
